data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataimport/log/SingleFileImportLogger.java

```diff
@@ -0,0 +1,230 @@
+          summaryLogWriter = createLogWriter(config.getLogDirectoryPath() + SUMMARY_LOG_FILE_NAME);
+        }
+      }
+    }
+  }
+
+  /**
+   * Writes the summary of a data chunk to the specified log writer.
+   *
+   * @param dataChunkStatus the status of the data chunk to log
+   * @param logWriter the log writer to write to
+   * @throws IOException if an I/O error occurs while writing to the log
+   */
+  private void writeImportDataChunkSummary(
+      ImportDataChunkStatus dataChunkStatus, LogWriter logWriter) throws IOException {
+    JsonNode jsonNode = OBJECT_MAPPER.valueToTree(dataChunkStatus);
+    writeToLogWriter(logWriter, jsonNode);
+  }
+
+  /**
+   * Gets the appropriate log writer for a transaction batch based on its success status. If the log
+   * writer doesn't exist yet, it will be created.
+   *
+   * @param batchResult the transaction batch result
+   * @return the log writer for the batch
+   * @throws IOException if an I/O error occurs while creating a new log writer
+   */
+  private LogWriter getLogWriterForTransactionBatch(ImportTransactionBatchResult batchResult)
+      throws IOException {
+    return batchResult.isSuccess() ? successLogWriter : failureLogWriter;
+  }
+
+  /**
+   * Writes the details of an import task result to the appropriate log files. Successful targets
+   * are written to success logs and failed targets to failure logs. The method is synchronized on
+   * the respective log writers to ensure thread safety.
+   *
+   * @param importTaskResult the result of the import task to log
+   * @throws IOException if an I/O error occurs while writing to the logs
+   */
+  private void writeImportTaskResultDetailToLogs(ImportTaskResult importTaskResult)
+      throws IOException {
+    for (ImportTargetResult target : importTaskResult.getTargets()) {
+      if (config.isLogSuccessRecords()
+          && target.getStatus().equals(ImportTargetResultStatus.SAVED)) {
+
+        writeToLogWriter(successLogWriter, OBJECT_MAPPER.valueToTree(target));
+      }
+      if (config.isLogRawSourceRecords()
+          && !target.getStatus().equals(ImportTargetResultStatus.SAVED)) {
+        writeToLogWriter(failureLogWriter, OBJECT_MAPPER.valueToTree(target));
+      }
+    }
+  }
+
+  /**
+   * Writes a JSON node to a log writer and flushes the writer.
+   *
+   * @param logWriter the log writer to write to
+   * @param jsonNode the JSON node to write
+   * @throws IOException if an I/O error occurs while writing
+   */
+  private void writeToLogWriter(LogWriter logWriter, JsonNode jsonNode) throws IOException {
+    logWriter.write(jsonNode);
+  }
+
+  /**
+   * Closes all log writers and sets them to null. This method is called when all data chunks have
+   * been completed.
+   */
+  private void closeAllLogWriters() {
+    synchronized (this) {
+      closeLogWriter(summaryLogWriter);
+      closeLogWriter(successLogWriter);
+      closeLogWriter(failureLogWriter);
+      summaryLogWriter = null;
```

Comment:
The reason we set `summaryLogWriter` to null is that we reopen the writers, right?

Reply:
Yes, it is set to null to so that if it need to be reinitialised it will be safe to do so.

Reply:
Are `successLogWriter` and `failureLogWriter` also reinitialized after being closed?
It looks like they are only initialized once in the constructor, and I don’t see this class being reused after calling `closeAllLogWriters()`.

Reply:
@brfrn169 san,

Yes you are right. We don't need the null assignment here. It was a mistake from my side as I had add null assignments (due to some previous conditions) to all 3 (succes, summary and failure) logs but missed to remove the summary one. You are also  right as in this scenario only one file is needed for a log type and once it is closed,there is no need to reopen it. Thank you for pointing it out. I have removed the null assignment.